# Name of the coordinate system
std_msgs/String name_cs

# Distance between the projector device and the projection surface
std_msgs/Float64 distance

# Real-world coordinates of reference points from the projector's reference system {P}
z_laser_projector/ReferencePoint p1
z_laser_projector/ReferencePoint p2
z_laser_projector/ReferencePoint p3
z_laser_projector/ReferencePoint p4

# Coordinates of the reference point T1 from the user's reference system {T}
z_laser_projector/ReferencePoint T1

# Resolution of the coordinate system
# The resolution is considered for the largest dimension of the coordinate system
# and it is calculated proportionally for the rest
std_msgs/Int16 resolution

---

z_laser_projector/UserCoordinateSystem T
std_msgs/Bool cs_created
std_msgs/String message